test/functional/JIT_Test/src/jit/test/jitt/array/multianewarray.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2001, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package jit.test.jitt.array;
+
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+import org.testng.Assert;
+
+@Test(groups = { "level.sanity","component.jit" })
+
+public class multianewarray extends jit.test.jitt.Test
+   {
+
+   private static Logger logger = Logger.getLogger(multianewarray.class);
+
+   private Integer[][] createMultiArray(int firstDimLength, int secondDimLength)
+      {
+      Integer[][] multiarray = new Integer[firstDimLength][secondDimLength];
+      for (int i = 0; i < firstDimLength; i++)
+         {
+         for (int j = 0; j < secondDimLength; j++)
+            {
+            multiarray[i][j] = i * j;
+            }
+         }
+      return multiarray;
+      }
+
+   private boolean testMultiArray(Integer[][] inputArray)
```

Comment:
We should also validate the lengths of the arrays are what we think they are, just in case the inline code doesn't allocate the arrays of correct sizes.

Reply:
The test methods now test for length as well: d825ed9